### ISSUE TITLE
Crash every hour

### ISSUE THREAD

**Issue by guanyingkai1984** [lmstudio-ai/lmstudio-bug-tracker#267]
We are experiencing significant performance issues with the API for Build 10 on Windows, even with smaller models (e.g., 3B). [I am using RTX4070]

Unpredictable Execution Times: The API exhibits highly variable execution times, with some predictions running indefinitely and ultimately causing LM Studio to crash. These prolonged hangs can last for hours.

Frequent Interruptions: To prevent these crashes, we are forced to constantly monitor predictions. If a prediction exceeds a reasonable threshold (e.g., 15 seconds), we need to manually intervene by adjusting parameters and reloading the model. This manual intervention disrupts workflow significantly.

**Comment by yagil** [comment]
Hello thanks for the report. 

Please share the following information:

- what errors are you getting exactly?
- what is crashing?  The model or LM Studio itself?
- are you able to share a way for us to reproduce this?

This information will be critical for us to be able to fix this. Thanks!

**Comment by guanyingkai1984** [comment]
![image](https://github.com/user-attachments/assets/d92798c8-ec20-4f16-9ab8-ec601819a95
![Screenshot 2024-12-28 161121](https://github.com/user-attachments/assets/3d9ac67e-f0e5-441e-9039-0feae6a2b343)


Please see the screenshot attached. 
It is stopped in the middle for an hour of doing nothing. 
Please see the time on the right hand side compare with the time of your app.

I use the API for load 20K row of data. It normally will stop and doing nothing / continue loading for hours.

**Comment by yagil** [comment]
OK so it doesn’t crash but it hangs?

Are you able to see what the model generates when it’s in this state?

**Comment by guanyingkai1984** [comment]
It will crash after handing for hours. e.g. today morning, the LM Studio is completely hang, and cannot see anything inside.

I do not keep the log of what model generates. And i think it doesn't generate anything. If it generate something, PY code will capture the generated stuff and go to next row.

**Comment by yagil** [comment]
When you can’t see anything inside, is the screen black? Or something else

How long after you start sending API requests does this happen?

**Comment by guanyingkai1984** [comment]
everything is normal , no black screen, if i amend the model config within a few hour after error.

That error happen quite randomly, sometime less than an hour, sometime a few hours.

**Comment by yagil** [comment]
Do you have access to the client program that makes the API requests?   Are you able to check what’s going on there when this happens?

**Comment by guanyingkai1984** [comment]
Yes. everything is on my same laptop.
Nothing wrong.
After i amend the config, the PY in the Jupyter Lab run normally again.

**Comment by yagil** [comment]
Which specific config amendment makes it work again?

**Comment by guanyingkai1984** [comment]
it will work if i can anything to load the model again.
(Because if i Eject the model, my PY code will have  error)
normally, i can switch on and off of the flash attention.

**Comment by yagil** [comment]
Here’s a link to b9: https://releases.lmstudio.ai/win32/x86/0.3.5/beta/9/LM-Studio-0.3.5-Setup.exe

Please let me know if you’re still running into the situation there.

**Comment by guanyingkai1984** [comment]
ok. Thanks. Let me try again after I finish this batch of PY.

**Comment by yagil** [comment]
@guanyingkai1984 are the results different?

**Comment by guanyingkai1984** [comment]
I reinstall the Build 9.
Still waiting for 'Bad' result

But today i have also installed the 'CUDA llama.cpp' I am not sure whether this 'help'.

**Comment by yagil** [comment]
Are you able to share instructions for us to try to reproduce the bad results?

**Comment by guanyingkai1984** [comment]
Single instruction doesn't reproduce bad results. I asked the same questions thousand and thousand of time will product bad result with Build 10, whether you will see the bad result depends on luck. I think you can generate a list of question to ask the news articles yourself. For each prompt i asked 8 different question and ask the LLM to reply answer.

I use Py code to loop all the news articles, each loop is around 2000 input token with less than 50 output token.

All the news from the following website:
https://oilprice.com/Latest-Energy-News/World-News/

**Comment by yagil** [comment]
Thanks, trying to reproduce this by sending an API request with ~2000 tokens in a loop.  Please let me know if there are any other factors worth mentioning.

Also, if you wanted and had time to go back to b10 and see if if still happens, that would be helpful

**Comment by guanyingkai1984** [comment]
I will try tomorrow, because i am still running with B9.

**Comment by guanyingkai1984** [comment]
![image](https://github.com/user-attachments/assets/1dcbe345-8e7c-41ed-bf1f-9f8d83322fdc)

I think B9 has the same problem.
Please see screenshot attached.

**Comment by yagil** [comment]
Interesting: wasn't B9 working for you previously?

Another point to consider- have you switched which model you're using for this recently?

**Comment by guanyingkai1984** [comment]
May be it worked for over 10K row for B8 for the same laptop. (May be you can let me download B8 to try again.)
May be the previous prompt is not as long as the current one, but it should not be significant. 

With the same laptop, i did run over 18K rows for the same input file without any problem, over 2 days, for a couples of times.
May be because recently you updated 'llama.cpp'

**Comment by yagil** [comment]
I think I was just able to reproduce this too with `hermes-3-llama-3.1-8b` specifically after about 70ish loops.

The model just goes into an extremely long generation, probably thrown off by the prompt. This might be solved with sampling parameter tweaks.

For comparison, I was NOT able to reproduce this in over 2000 loops with `llama-3.2-3b-instruct`.

Are you able to control which model you're using?  I don't think this is a new problem in LM Studio B10.

**Comment by guanyingkai1984** [comment]
Thanks for your reply. 
May be it is the old problem.

I think i tried to use Qwen2.5 7B before, it has the same problem.
To me, llama is the best and Qwen is the 2nd best model to use.

And i think i need to use 8Billion model at least, 3B model is not good enough. This problem happen by chance, my previous one has problem over 10K loops. But as long as you know there is a problem, that's good.

Thanks again.

**Comment by guanyingkai1984** [comment]
Hello, I ran meta-llama-3.1-8b-instruct for all 18K rows of data with Build 9, encountering no errors. This behavior may be model-dependent.

**Comment by guanyingkai1984** [comment]
![Screenshot 2025-01-01 072621](https://github.com/user-attachments/assets/1ddf5d65-32e2-4549-a639-065c38b3b14e)

B10, Mistral down for a few hours.
Please see the screenshot,this time, the model is rejected itself, to me that is worst
